test/cpp/cpp_test_util.cpp

```diff
@@ -25,11 +46,13 @@ at::Tensor ToCpuTensor(const at::Tensor& t) {
   return xtensor ? ToTensor(*xtensor) : tensor;
 }
 
-bool EqualValues(at::Tensor tensor1, at::Tensor tensor2) {
-  if (tensor1.sizes() != tensor2.sizes()) {
-    std::cerr << "Different sizes:\n"
-              << tensor1.sizes() << "\n-vs-\n"
-              << tensor2.sizes() << "\n";
+bool EqualValues(at::Tensor tensor1, at::Tensor tensor2,
```

Comment:
Do not overload this one. Add a new one with proper name, that also does the element type check.

Reply:
Sounds good, the code duplication would be minor.

Reply:
Done it the other way around since we almost always want to check the element type, only cast operations are different.